### PR TITLE
update feature store arm template to support user provided default st…

### PR DESCRIPTION
**Pull request by runhli**
…orage account

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 1]
```diff
@@ -647,17 +648,17 @@
...
             "type": "Microsoft.Storage/storageAccounts",
             "apiVersion": "2019-04-01",
-            "name": "[parameters('storageAccountName')]",
+            "name": "[variables('provisionedStorageAccountName')]",
             "tags": "[parameters('tagValues')]",
             "dependsOn": [
                 "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('subnetName'))]"
             ],
-            "location": "[parameters('storageAccountLocation')]",
+            "location": "[if(equals(parameters('set_up_feature_store'), 'false'), parameters('storageAccountLocation'), parameters('location'))]",
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:659]: Can you change it to:
if equals('set_up_feature_store', true), parameters('location'), parameters('storageAccountLocation')

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 2]
```diff
@@ -647,17 +648,17 @@
...
             "sku": {
-                "name": "[parameters('storageAccountType')]"
+                "name": "[if(equals(parameters('set_up_feature_store'), 'false'), parameters('storageAccountType'), 'Standard_LRS')]"
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:661]: same here. Change the condition to equals(xxx, 'true')

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 3]
```diff
@@ -674,18 +675,19 @@
                 },
                 "supportsHttpsTrafficOnly": true,
                 "allowBlobPublicAccess": false,
-                "networkAcls": "[if(equals(parameters('storageAccountBehindVNet'), 'true'), variables('networkRuleSetBehindVNet'), json('null'))]",
+                "networkAcls": "[if(and(equals(parameters('storageAccountBehindVNet'), 'true'), equals(parameters('set_up_feature_store'), 'false')), variables('networkRuleSetBehindVNet'), json('null'))]",
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:678]: should offline store container also config networkAcls?

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:678]: it's always false as before

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json (hunk 1 above)
Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json]: done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json (hunk 2 above)
Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json]: done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json (hunk 3 above)
Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json]: removed set up feature store condition

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 4]
```diff
@@ -647,17 +648,17 @@
             "location": "[parameters('location')]"
         },
         {
-            "condition": "[and(variables('enablePE'), equals(parameters('storageAccountOption'), 'new'))]",
+            "condition": "[and(variables('enablePE'), or(equals(parameters('storageAccountOption'), 'new'), and(equals(parameters('set_up_feature_store'), 'true'), empty(parameters('offline_store_target')))))]",
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:651]: do we need the parameter of "set_up_feature_store"? can we use "kind" parameter for this?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 5]
```diff
@@ -647,17 +648,17 @@
...
             "type": "Microsoft.Storage/storageAccounts",
             "apiVersion": "2019-04-01",
-            "name": "[parameters('storageAccountName')]",
+            "name": "[variables('provisionedStorageAccountName')]",
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:654]: Is this variable needed? Is not it just the parameter('storageAccountName')

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:654]: this is needed, it represents the storage account name that needs to be provisioned, parameter('storageAccountName') could be user provided storage account name.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json (hunk 4 above)
Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:651]: we can use either of them.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 6]
```diff
@@ -688,18 +678,19 @@
                 },
                 "supportsHttpsTrafficOnly": true,
                 "allowBlobPublicAccess": false,
-                "networkAcls": "[if(equals(parameters('storageAccountBehindVNet'), 'true'), variables('networkRuleSetBehindVNet'), json('null'))]",
-                "isHnsEnabled": "[equals(parameters('set_up_feature_store'), 'true')]",
-                "minimumTlsVersion": "TLS1_2"
+                "networkAcls": "[if(and(equals(parameters('storageAccountBehindVNet'), 'true'), equals(parameters('storageAccountOption'), 'new')), variables('networkRuleSetBehindVNet'), json('null'))]",
```
Comment (qjxu) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:681]: I think networkAcls should apply when: (1) the storage is only for default blob; (2) the storage is for both default blob and offline store.

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:681]: yes, I think that's current behavior

Comment (qjxu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:681]: Current code:
when not equals(parameters('storageAccountOption'), 'new'))

the network ack is json('null').

But the original behavior is different.

Comment (runhli) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:681]: when not equals(parameters('storageAccountOption'), 'new')), this is feature store's offline store storage provision scenario, and it's a separate storage other than default storage. we should not apply networkAcls, which matches (2)

**Review by qjxu — APPROVED** [review]

**Review by bastrik — APPROVED** [review]

**Pull request merged**